component/control-api.jsonnet

```diff
@@ -0,0 +1,392 @@
+);
+
+local countries_yaml = std.manifestYamlDoc(countryList);
+
+local hasCountriesConfig = params.odoo8.countries != null && std.length(params.odoo8.countries) > 0;
+
+/////////////////
+// API Server
+
+local validCertSecret =
+  if params.apiserver.tls.certSecretName != null then
+    assert std.length(params.apiserver.tls.serverCert) > 0 : 'apiserver.tls.serverCert is required';
+    assert std.length(params.apiserver.tls.serverKey) > 0 : 'apiserver.tls.serverKey is required';
+    true
+  else
+    false
+;
+
+local apiserverDeploymentArgs =
+  [
+    '--invitation-storage-backing-ns=' + params.invitation_store_namespace,
+  ]
+  +
+  (if validCertSecret then
+     [
+       '--tls-cert-file=/apiserver.local.config/certificates/tls.crt',
+       '--tls-private-key-file=/apiserver.local.config/certificates/tls.key',
+     ]
+   else
+     [])
+  +
+  (if hasCountriesConfig then
+     [
+       '--billing-entity-odoo8-country-list=/config/billing_entity_odoo8_country_list.yaml',
+     ]
+   else
+     [])
+  + params.apiserver.extraArgs
+;
+
+local apiserverDeploymentArgsPatch = {
+  patches+: [
+    {
+      patch: std.format(|||
+        - op: add
+          path: "/spec/template/spec/containers/0/args/-"
+          value: "%s"
+      |||, arg),
+      target: {
+        kind: 'Deployment',
+        name: 'control-api-apiserver'
+      },
+    },
+    for arg in apiserverDeploymentArgs
+  ],
+};
+
+local apiserverOdooConfigPatch = if hasCountriesConfig then {
+  patches+: [
+    {
+      patch: std.format(|||
+        - op: add
+          path: /spec/template/spec/containers/0/volumeMounts/-
+          value:
+            mountPath: /config/billing_entity_odoo8_country_list.yaml
+            name: countries-config
+            readOnly: true
+            subPath: billing_entity_odoo8_country_list.yaml
+        - op: add
+          path: /spec/template/spec/volumes/-
+          value:
+            name: countries-config
+            configMap:
+              name: billing-entity-odoo8-country-list
+        - op: add
+          path: /spec/template/metadata/annotations
+          value:
+            'checksum/countries': %s
+      |||, std.md5(countries_yaml)),
+      target: {
+        kind: 'Deployment',
+        name: 'control-api-apiserver'
+      },
+    },
+  ],
+} else {};
+
+local apiserverExtraEnvList = com.envList(params.apiserver.extraEnv);
+local apiserverDeploymentEnvPatch = if std.length(apiserverExtraEnvList) > 0 then {
+  patches+: [
+    {
+      patch: std.format(|||
+        - op: add
+          path: /spec/template/spec/containers/0/env
+          value: []
+        - op: add
+          path: /spec/template/spec/containers/0/env/-
+          value: %s
+      |||, apiserverExtraEnvList),
+      target: {
+        kind: 'Deployment',
+        name: 'control-api-apiserver'
+      },
+    },
+  ],
+} else {};
+
+local apiserverDeploymentResources = std.mergePatch({
+  limits:{
+    cpu: "300m",
+    memory: "100Mi"
+  }, 
```

Comment:
There are some trailing whitespaces in this file, see the failing editorconfig test.
A `make format`, should resolve the issues.